app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+  after_save :update_post_counter
+
+  def update_post_counter
+    author.increment!(:posts_counter)
+  end
```

Comment:
As requested by the previous reviewer, please make sure that the 3 methods for updating counters (`update_post_counter`, `update_likes_counter`, `update_comments_counter`) are protected by placing them below the `private` keyword.

- This will make sure that these methods are not accessible anywhere in the app. Since these methods should be strictly triggered only when a user creates a post, comment, or like, only the `after_save` Rails Callback should have access to these methods in order to prevent inconsistent data.